src/task.c

```diff
@@ -1043,6 +1043,119 @@ JL_DLLEXPORT void jl_gdb_dump_threadinfo(void)
 }
 #endif
 
+
+#ifdef _OS_LINUX_
+#if defined(__i386__)
+static uintptr_t ptr_mangle(uintptr_t p)
```

Comment:
This seems mysterious! It would be nice to have a comment describing what `ptr_mangle` does (eg just linking to the relevant platform ABI spec?)

Reply:
There is no documentation on this. It's just some secret magic values taken from elsewhere.